verifier/detail_template.html

```diff
@@ -224,11 +224,41 @@
         const test_data = test_results[box_class];
         test_data.selected_set = null;
 
-        // TODO: Finish this
+        // TODO: Update to handle tristate
+        let excluded_set = new Set();  // Remove these after the loop.
         let characterized_labels = test_data.box_labels;
         for (let check_box of test_data.check_boxes) {
+
             // For each checkbox
-            if (check_box.checked) {
+      if (check_box.attributes.array_labels) {
+      // This is a tristate item.
+      const which_state = check_box.value;
+      
+      // Choose the value depending on the status of the control
+      const labels_string = check_box.attributes.array_labels.value;
+      const str_len = labels_string.length;
+      const labels = labels_string.split(",");
+      const newSet = new Set(labels);
+      if (which_state == cross_mark) {
+      for (let label of labels) {
+      excluded_set.add(label);
+      }
+      continue;
+      }
+      if (which_state == question_mark) {
+      continue;
+      }
+      if (! test_data.selected_set) {
+      // Start the set of selected items.
+      test_data.selected_set = newSet;
+      } else {
+      // Update by intersection with this set.
+      test_data.selected_set =
+      new Set([...test_data.selected_set].filter((x) => newSet.has(x)));
+      }
+      }
+            else if (check_box.checked) {
+               // A checked box.
```

Comment:
fix the indentation of this region

Reply:
Fixed this. Also removed old code and revised a lot to be clearer. PTAL.